processor/hostmetricsstackdriverprocessor/hostmetrics_stackdriver_processor_test.go

```diff
@@ -0,0 +1,339 @@
+		}
+
+		// assert equality of IL metrics
+		ilmsAct := rmAct.InstrumentationLibraryMetrics()
+		ilmsExp := rmExp.InstrumentationLibraryMetrics()
+		require.Equal(t, ilmsExp.Len(), ilmsAct.Len())
+		for j := 0; j < ilmsAct.Len(); j++ {
+			ilmAct := ilmsAct.At(j)
+			ilmExp := ilmsExp.At(j)
+
+			// currently expect IL to always be nil
+			assert.True(t, ilmAct.InstrumentationLibrary().IsNil())
+			assert.True(t, ilmExp.InstrumentationLibrary().IsNil())
+
+			// assert equality of metrics
+			metricsAct := ilmAct.Metrics()
+			metricsExp := ilmExp.Metrics()
+			require.Equal(t, metricsExp.Len(), metricsAct.Len())
+
+			// build a map of expected metrics
+			metricsExpMap := make(map[string]pdata.Metric, metricsExp.Len())
+			for k := 0; k < metricsExp.Len(); k++ {
+				metricsExpMap[metricsExp.At(k).MetricDescriptor().Name()] = metricsExp.At(k)
+			}
+
+			for k := 0; k < metricsAct.Len(); k++ {
+				metricAct := metricsAct.At(k)
+				metricExp, ok := metricsExpMap[metricAct.MetricDescriptor().Name()]
+				if !ok {
+					require.Fail(t, fmt.Sprintf("unexpected metric %v", metricAct.MetricDescriptor().Name()))
+				}
+
+				// assert equality of descriptors
+				assert.Equal(t, metricExp.MetricDescriptor(), metricAct.MetricDescriptor())
+
+				// assert equality of int data points
+				idpsAct := metricAct.Int64DataPoints()
+				idpsExp := metricExp.Int64DataPoints()
+				require.Equal(t, idpsExp.Len(), idpsAct.Len())
+				for l := 0; l < idpsAct.Len(); l++ {
+					idpAct := idpsAct.At(l)
+					idpExp := idpsExp.At(l)
+
+					assert.Equal(t, idpExp.LabelsMap().Sort(), idpAct.LabelsMap().Sort())
+					assert.Equal(t, idpExp.StartTime(), idpAct.StartTime())
+					assert.Equal(t, idpExp.Timestamp(), idpAct.Timestamp())
+					assert.Equal(t, idpExp.Value(), idpAct.Value())
+				}
+
+				// assert equality of double data points
+				ddpsAct := metricAct.DoubleDataPoints()
+				ddpsExp := metricExp.DoubleDataPoints()
+				require.Equal(t, ddpsExp.Len(), ddpsAct.Len())
+				for l := 0; l < ddpsAct.Len(); l++ {
+					ddpAct := ddpsAct.At(l)
+					ddpExp := ddpsExp.At(l)
+
+					assert.Equal(t, ddpExp.LabelsMap().Sort(), ddpAct.LabelsMap().Sort())
+					assert.Equal(t, ddpExp.StartTime(), ddpAct.StartTime())
+					assert.Equal(t, ddpExp.Timestamp(), ddpAct.Timestamp())
+					assert.Equal(t, ddpExp.Value(), ddpAct.Value())
+				}
+
+				// currently expect other kinds of data points to always be empty
+				assert.True(t, metricAct.HistogramDataPoints().Len() == 0)
+				assert.True(t, metricExp.HistogramDataPoints().Len() == 0)
+				assert.True(t, metricAct.SummaryDataPoints().Len() == 0)
+				assert.True(t, metricExp.SummaryDataPoints().Len() == 0)
+			}
+		}
+	}
+}
+
+var cached *pdata.Metrics
+
+// a very dirty hack to get an internal.Data object since we shouldn't be using it yet,
+// but its needed to construct the resource tests
+func newInternalMetrics() pdata.Metrics {
```

Comment:
This is an extremely dirty hack to get metrics using the internal data format. Writing this translation code using the old data format would have added considerable complexity due to the batches of resource data being processed separately (therefore meaning we would need to store state and re-combine the batches).

This code configures the host metrics receiver to generate metrics, which we know are in the new format, and gets this returned as an opaque `pdata.Metrics` object we can use. We do this once, and then Clone the empty object to create multiple new internal metrics objects.

Fortunately this only effects the tests. I will clean this up once the new data format is finalized and properly released.